### ISSUE TITLE
[BUG] Instance has been ratelimited 

### ISSUE THREAD

**Issue by eddydc** [benbusby/whoogle-search#707]
**Describe the bug**
Getting 

Error

Instance has been ratelimited

[Continue your search with Farside](https://farside.link/)
Whoogle:
https://farside.link/whoogle/search?q=dit is een test&lang_interface=lang_en&country=BE&theme=system

Searx:
https://farside.link/searx/search?q=dit is een test

[Return Home](https://synoogle:Hf4LhYGvsg@wilma.idizaai.be/home)

**To Reproduce**
Just do a search with Whoogle

**Deployment Method**
- [ ] Docker-compose

**Version of Whoogle Search**
lastest

**Desktop (please complete the following information):**
Win 10, latest FF

**Comment by T3chHunt3r** [comment]
I too have this issue on version 0.7.0

**Comment by ShlomiD83** [comment]
same issue for me.
can't resolved the issue no matter what I do.

**Comment by sysoppl** [comment]
Same for me, just deployed and it's not working at all.

**Comment by ironboy1** [comment]
Same on my side for version 0.7.1 and 0.7.2

**Comment by bruvv** [comment]
This has been answered already: https://github.com/benbusby/whoogle-search/issues/211

**Comment by pieslinger** [comment]
I am having rate limiting issues occasionally and I am the only user of whoogle running in a container. I do not see how #211 has answered this already. I would install this for more people in my life as well if this issue goes away. Is there just something I need to modify in the setup?

**Comment by pieslinger** [comment]
I also only noticed it in 0.7.1 and 0.7.2 - although 0.7 was my first time using this. Just installed 0.7.3 tonight.

**Comment by aw-jansen** [comment]
Same here. Getting the rate limiting message when connected via a NordVPN container. Would it still be a possibility to implement a workaround? On a different topic someone mentioned changing the query: https://github.com/searxng/searxng/issues/159

**Comment by haorenxu** [comment]
Same here on 0.7.4 using a public IP with password. Getting limited for average one or two days. Restarting the docker can fix it, but the problem always appears again.

**Comment by jojolll** [comment]
I use my own Whoogle instance (alone). Almost every day I reach the ratelimit after 10 to 15 searches during the day. 

I have the change with my LiveBox to be able with a simple click in the interface to renew my IP, that solves the problem for 1-2 days. 
It's still very annoying and makes the solution of a personal Whoogle server very unviable.

**Comment by Albonycal** [comment]
there isn't much you can do about this issue, google blocks the requests it thinks are suspicious. If your IP is dynamic, try rebooting or change your VPS's IP.

**Comment by dr460nf1r3** [comment]
This might in fact also be caused by bots crawling the instances. At least I noticed that the rate limits became less after blocking most known bots via Cloudflare WAF (12k events over 24h) 🤔

**Comment by bruvv** [comment]
For me. A restart of the docket container fixes this.

**Comment by whaler-ragweed** [comment]
> Same here on 0.7.4 using a public IP with password. Getting limited for average one or two days. Restarting the docker can fix it, but the problem always appears again.

I moved my VPS form Linode to Amazon Lightsail 5 months ago and never have the issue again.

**Comment by ForceConstant** [comment]
It is not clear from the error message, but looking at the code, the ratelimit is external to whoogle, and it is coming from a service that whoogle is querying. And if it gets a captcha, then it reports this.  So I couldn't imaging how @bruvv restarting your container would affect this.

**Comment by 0xspade** [comment]
> For me. A restart of the docket container fixes this.

Doesn't work on my side.

**Comment by CptanPanic** [comment]
For me, it turns out it was the proxy I was using. I guess there was too
many people at that IP address.

**Comment by Paulie420** [comment]
So... I get this with both whoogle-search and googler - only when connected to my Private Internet Access VPN on some of their locations. Canada being one of them.... I'd bet a lot of folks seeing this issue are simply behind a VPN that Google is blocking - if this describes your setup, try disconnecting from your VPN and running whoogle-search again; viola, fixed.

**Comment by dr460nf1r3** [comment]
From my understanding it's the IP getting rate-limited, so using another one should work.

**Comment by Snake16547** [comment]
Easy to say to change the IP when you have a VPS

**Comment by brokoler** [comment]
Since the last 2 days all of the public Whoogle instances are affected by this, most times only one of them works and yesterday it happened that even with Farside all instances were rate limited.

To me it seems like Google is going against Whoogle, but I still hope it's just a bug in the latest Whoogle software that can be fixed.

**Comment by Snake16547** [comment]
I switched to the Oracle free tier and until now I'm free of any ratelimit messages.

**Comment by benbusby** [comment]
@brokoler public instances go through waves of all being rate limited, and then being clear again. It looks like some are available now. There's no simple fix to get around it unfortunately, but it isn't a bug in Whoogle. I always recommend using your own Whoogle instance, or running it locally on your machine (behind a VPN if you want) if rate limiting becomes too much of a problem.

**Comment by MaverickTN** [comment]
I just had this issue for the first time after updating to 0.8.2. I haven't used it in 3 days, and it's ratelimited on all queries. Restarted the container, no  change.

**Comment by Johnnii360** [comment]
I'm joining this issue. Same problem on my instance. Maybe it could be a workaround to use an API key in the developer console but it has to be implemented by @benbusby . It's a common thing that Google shows an CAPTCHA request if there too many search request of one and the same IP. I think it's their "DDoS protection".

**Comment by robertpartridge** [comment]
I've run into this problem today with my deployment of version 0.8.2.  My instance is behind basic HTTP auth with a fairly strong unique password so that it's not open to just anyone to use.  Any advice?

**Comment by seirdotexe** [comment]
Getting this too now on my own IP address, locally hosted. Already restarted the container and even reinstalled it.

**Comment by akovia** [comment]
Just installed whoogle 0.8.2-1 for the first time ever from Arch aur, so have no frame of reference. how it's supposed to even work.

Is there anything I can do to get this working? 
I'm behind a VPN, so is that an actual issue to run a self hosted instance?

**Comment by imTHAI** [comment]
Today I face the same problem. But I use it on an internal docker container, and I'm the only one who use it in the family. How is it possible that I have reached the rate limit of search requests for google, from a single home user ?!

**Comment by winhex** [comment]
Same problem. I'm deploying the docker container with whoogle 0.8.4 at my own home server. No public access, only personal using. Instance has been rate limited after 20-30 min of using

**Comment by Drallas** [comment]
I also have been rate limted, my Whoogle runs in a tteck LXC container. Updating the OS doesn't solve it!

**Comment by zidanpragata** [comment]
whatever time, if direct search method is not changed to the brilliant idea of "yacy search", error message "rate is limited" will still occur frequently.. the easiest solution to overcome this is to change the IP.. 

 failed :)

**Comment by aindriu80** [comment]
I tried changing IP and using VPN but still getting rate limited, very frustrating

**Comment by akovia** [comment]
I bit the bullet and signed up for Kagi now that they made unlimited searches more affordable. I find it better than Google in so many ways and doubt I'll ever go back. I have no skin in the game, but they have a free trial if you want to check it out.

Either way I hope you get it figured out.

**Comment by Snake16547** [comment]
I had it running on Oracles free tier with zero problems. Several clients used my instance.
Now I switched to Hetzner Cloud and running smoothly as well.

**Comment by imTHAI** [comment]
Could someone explain why a whoogle instance can be sometimes rate-limited when the other computers (behind the same public ip) aren't when using google ? 
Are the requests from whoogle so special (and harmful?) that Google identifies them and limits them ?

**Comment by Drallas** [comment]
> I bit the bullet and signed up for Kagi now that they made unlimited searches more affordable. I find it better than Google in so many ways and doubt I'll ever go back. I have no skin in the game, but they have a free trial if you want to check it out.
> 
> Either way I hope you get it figured out.

Whoogle is nice, but the search results from Google feel manipulated and without still optimized to show payed results first. Would be better if Whoogle could include more search options. Kagi might be an alternative too!

**Comment by MuubNito** [comment]
*ALL* instances i am trying or doesnt work or just straight up give me another issue... Ouch

**Comment by imTHAI** [comment]
I don't know about you, but I can use my whoogle container again without any problems. 
Maybe google changed something on his side ?

**Comment by seirdotexe** [comment]
My instance is getting rate limited constantly now for some reason, only after 20 searches. Makes it impossible to use Whoogle

**Comment by C8opmBM** [comment]
happened to me as well 2 times today.

**Comment by EasternPA** [comment]
I just got this for the first time ever. I am the sole user of a docker-compose stack on Oracle Free Tier. I recently lost SSH access due to rebuilding my PC, so I need to rebuild the cloud instance anyway. Perhaps a rebuild will clear it up.

**Comment by Zundrium** [comment]
Same here, CHAPTA seems to be the issue

**Comment by mlause** [comment]
Same here. Is there anyway to passthrough the captcha?

**Comment by working-name** [comment]
I run whoogle both behind a residential connection and datacenter (so I can switch between them when one gets rate limited). 

There's **no difference** in getting rate limited **based on IP** or IP type (residential vs data center). 
I'm the only one using both instances, checked logs. 

I do aggressively google things though (still manual, not automated, so definitely not more than 1 search per second), sometimes with minor tweaks (like "sle crescentic" which later changed to "psgn crescentic") and that's when I notice a higher incidence of rate limiting.

**Comment by jobsidi** [comment]
Got this error. Logs: ERROR:app:503 (CAPTCHA)
searched last time 6 monthes ago

**Comment by eyalzek** [comment]
I'm seeing the same (`ERROR:app:503 (CAPTCHA)`) on a newly deployed self-hosted instance after about 10 queries...
Is there an obvious cause for that? Any way to "reset" it? Any configuration best avoided?

**Comment by boognish-rising** [comment]
I just got rate limited on a local install (not hosted, available only via http://localhost:5000). I was completely caught off guard, as I've used whoogle locally for years without getting rate limited. Like @byertsy, I was checking out different CSS themes and submitted the same search of "test" maybe 10 times when it occured. I wonder if it was repeating the same search or perhaps doing so with a different css theme each time and thus, a different URL that triggered it?

Here's hoping it was an isolated incident

**Comment by martijnboers** [comment]
I have a feeling adjusting settings seem to trigger this? I was querying yesterday all day fine but now after changing things like search language and country getting rate-limited..

**Comment by martijnboers** [comment]
Querying fine again for multiple days and not touching any of the settings.
